### PR TITLE
Add `use-t-well` rule

### PR DESCRIPTION
**Pull request by jfmengels**
Adds `use-t-well` rule

I originally wanted to make the rule `no-unknown-assertions`, but adding checks to member expressions (`t.context`, ...) was easy, so I've made the rule more global. It now handles quite a few cases. It looks fine to me now, but I wouldn't be surprised if I missed some edge cases (false positives mostly). (If you guys have a a few projects to test it on, that'd be nice)

Let me know what you think :)

### CODE REVIEW

File: docs/rules/use-t-well.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Prevent the incorrect use of `t`
+
+Prevent the use of unknown assertion methods and the access to members other than the assertion methods and `context`, as well as some known misuses of `t`.
+
+
+## Fail
+
+```js
+import test = from 'ava';
```
Comment (novemberborn) [inline docs/rules/use-t-well.md]: Illegal `=` there :wink:

Comment (jfmengels) [inline docs/rules/use-t-well.md]: Just checking whether you guys actually read the diffs *whistle*

Comment (novemberborn) [inline docs/rules/use-t-well.md]: Nah I just lint them :stuck_out_tongue_winking_eye:

**Comment by sindresorhus** [comment]
No comments. This is perfect. Seriously. 🦄

**Comment by jfmengels** [comment]
:D
